src/coreclr/hosts/coreshim/CoreShim.cpp

```diff
@@ -64,10 +64,10 @@ namespace
         return std::wstring{ buffer.Buf, buffer.Buf + len };
     }
 
-    std::wstring GetEnvVar(_In_z_ const WCHAR *env)
+    std::wstring GetEnvVar(_In_z_ const WCHAR *env, _In_opt_ bool throwOnEmpty = true)
```

Comment:
I realize this is a common idiom in the CoreCLR codebase but in my opinion it is a very bad design. Either a function throws or it doesn't. Instead please create another function that uses try semantics returns an `HRESULT`. For example:

```
bool TryGetEnvVar(_In_z_ const WCHAR *, _Inout_ std::wstring &);
```

This could be implemented any number of ways, but doesn't have the default argument nor do arguments dictate function semantics.